packages/js/currency/src/test/utils.ts

```diff
@@ -0,0 +1,285 @@
+/**
+ * External dependencies
+ */
+import { numberFormat } from '@woocommerce/number';
+
+/**
+ * Internal dependencies
+ */
+import {
+	localiseMonetaryValue,
+	unformatLocalisedMonetaryValue,
+} from '../utils';
+
+const config = {
+	code: 'USD',
+	symbol: '$',
+	symbolPosition: 'left',
+	decimalSeparator: '.',
+	priceFormat: '%1$s%2$s',
+	thousandSeparator: ',',
+	precision: 2,
+};
+
+describe( 'localiseMonetaryValue', () => {
+	it( 'should format a number input correctly', () => {
+		expect( localiseMonetaryValue( config, 1234 ) ).toBe(
+			numberFormat( config, 1234 )
+		);
+	} );
+
+	it( 'should format string numbers correctly', () => {
+		expect( localiseMonetaryValue( config, '123456789' ) ).toBe(
+			'123,456,789.00'
+		);
+	} );
+
+	it( 'should format with swapped decimal and thousand separator', () => {
+		const customConfig = {
+			...config,
+			decimalSeparator: ',',
+			thousandSeparator: '.',
+		};
+		expect( localiseMonetaryValue( customConfig, '123.456.789' ) ).toBe(
+			'123.456.789,00'
+		);
+	} );
+
+	it( 'should not format incorrectly formatted numbers according to current config', () => {
```

Comment:
[Non-blocking; opinionated]: Since we've already been concerned about failing silently, should we throw an error for incorrect thousand separator placements? 

I'm working of a personally biased understanding of how I would work with WC, but if I add a thousand separator, then I'd expect it to be in the correct location, so if I wrote `7,50` then I'd prefer to get an error saying it doesn't look like an expect format (which should probably have been `7,500`).
That being said then I could also have mistyped `75,00` which should be `7500` so ignoring it would be correct in this case; but is that enough to counterweight allowing the other case?

Reply:
This function actually formats the value retrieved from the backend, for display in the frontend - I think it's correct to ignore incorrectly formatted numbers so that it throws an error when the user saves it. We don't currently have a routine that displays an error thrown where this function is called 🥲

This is out of consideration that we might have mal-formatted numbers saved prior to this PR, which should not occur going forward.